app/controllers/songs_controller.rb

```diff
@@ -0,0 +1,46 @@
+class SongsController < ApplicationController
+
+  def index
+
+  end
+
+  def new
+
+  end
+
+  def show
+    client = Grooveshark::Client.new({session: session[:groove_session]})
+    @groove_song = Song.find(params[:id])
+    @groove_url = client.get_song_url_by_id(@groove_song.link)
+  end
+
+  def create
+    song = Song.new(title: params[:song_name], artist: params[:song_artist], link: params[:song_id], playlist_id: params[:playlist_id], coverart: params[:coverart])
```

Comment:
Why aren't you using strong_parameters here?  I see why in the PlaylistController, but this would be much cleaner if it was just `Song.new(song_params)`